### DISCUSSION TITLE
Precedence of GEOL_DESC in 'Simplified GEOL' versus 'Extended GEOL' using Component description fields

### DISCUSSION THREAD

**Discussion by RobDayArup** [australiangeomechanics/agsformat#71] [Q&A]
Please note that this post is intended to be constructive and is not a criticism.  We were requested to post these types of issues to give others opportunity to respond.  Please comment on whether you agree/disagree.  I've done this as Q&A so answers are enabled.  perhaps a vote could be set up depending on responses.
 
I do not agree with the comments made by the AGS4.1AU1.2 Committee at the AGS roadshow 22/3/2023 that:
•	Best practice for entering GEOL is all or nothing – you either enter a full non component description directly into GEOL_DESC (Simplified GEOL), or you fully enter the description using component fields
•	Where component fields are present in the AGS file they take precedence over the description in GEOL_DESC.

I believe it contradicts the advice from the Committee as posted on  https://australiangeomechanics.org/ags-4-1-au/ that:
“The geological description is always consolidated under a single heading (GEOL_DESC) in this group and the individual ‘granular’ headings are optional as part of an extended GEOL group, depending upon the amount of granularity preferred for a particular project.”  

I also believe adopting this “best practice” would result in ambiguity in the AGS files, discourages contractors/consultants producing AGS data from providing the “Extended GEOL” and instead encourages them to provide the “Simplified GEOL” form of the AGS file whenever they can.

I think that using Component descriptions is highly desirable and contractors/consultants should be encouraged to provide ANY component fields that have used whether they completely describe the description or not.  Even for a contractor/consultant that elects to use the Extended GEOL format there will be many instances where data will contain only a partial description in Component form insufficient to build the full description at it appears on a log such that only partial AGS component data is available.  There are lots of reasons why this may be the case including where:
•	the AGS file contains information from different historical drilling campaigns,
•	the AGS file was prepared from PDF logs to only show sufficient information for generating cross section fences,  or
•	the contractor/consultant has decided to go down the Simplified GEOL route but has partially documented some fields to assist with data validation, preparing cross-sections or querying data.

Remembering that the AGS4.1 AU1.2 format is software agnostic It can be taken as a given that any contractor/consultant or software provider who is using component descriptions has written code in the software to build a description of GEOL_DESC.  However, because the rules for writing this code are not prescribed in AGS41.AU1.2, even if you have entered all of the Component fields, the resulting description prepared by different software could be subtly different.  Hence, I agree with the Aus Geomechanics Society website advice that the consolidated description is ALWAYS saved in GEOL_DESC.

BUT because of this, there will always be a GEOL_DESC in every data entry and there may be Component fields that do or do not represent the whole description.  In these cases it is impossible for the person/software receiving the data to know if the information in GEOL_DESC was intended to be completely built from the component fields or was entered as a non-component description. 

**I propose the following clarification and amendment to address this:
•	GEOL_DESC is always populated in the AGS file and is the definitive geological description identical to the description as it appears on the logs.  This is the description that takes precedence irrespective of whether component description fields are filled in or not.  
•	A new boolean Y/N field GEOL_ISCD needs to be added to GEOL to indicate whether the data in GEOL_DESC is a non-component description or can be COMPLETELY defined by the component description. An ‘N’ would indicate it is a non-component description and any component fields are for supplementary information.**

To ensure no data loss, when importing AGS files GEOL_DESC needs to be imported as a static text field that is not rebuilt by any software code and provides a record of the received data.  The naming of this field internally is up to the software, but for ease of discussion below I will call it GEOL_DDES (“driller’s description”  - arguably corresponds to DLOG_DESC but in the GEOL group so it has same depths as legend and origin).

When internally processing and exporting GEOL_DESC AGS data in the software the following needs to be taken into account depending on the value of GEOL_ISCD:
If GEOL_ISCD=N
•	Any internal software used to compile a GEOL_DESC from component fields is disabled; and
•	GEOL_DESC is populated with GEOL_DDES for internal and AGS outputting purposes.
•	All other GEOL fields are input and output unchanged in AGS.
If GEOL_ISCD=Y
•	Software is used to compile GEOL_DESC from the component fields (if the software has this ability)
•	The result is compared with GEOL_DDES (ideally they would be exactly the same, but probably wont be unless software used to compile the components is identical)
•	If the two are different, the database manager needs to make a decision on whether GEOL_DESC or GEOL_DDES is used to populate GEOL_DESC in output AGS files.  This decision will depend on who has ultimate responsibility for the accuracy of the data.
•	All other GEOL fields are input and output unchanged in AGS. 
  
Please let me know your thoughts

**Comment by SolidRockProgrammer** [comment]
Thanks Rob, the Working Group needs some time to discuss your suggestions.

**Comment by digitalgeotechnics** [comment]
It's worth noting WRT 

>I believe it contradicts the advice from the Committee as posted on https://australiangeomechanics.org/ags-4-1-au/ that:
“The geological description is always consolidated under a single heading (GEOL_DESC) in this group and the individual ‘granular’ headings are optional as part of an extended GEOL group, depending upon the amount of granularity preferred for a particular project.”

the `GEOL` table guidance notes include
> * Material descriptions shall be input in the GEOL_DESC as a combined description, and optionally split in the individual component description headings (as agreed with the client).
>* GEOL_DESC must be populated even if the individual component description headings are populated.
>* If component description headings are input, they shall take precedence over the combined GEOL_DESC.

**Reply by RobDayArup** [reply]
H i- where are the "GEOL table guidance notes" saved?

**Comment by SolidRockProgrammer** [comment]
They are in the excel data dictionary, on Group sheet, column J "Australia - Notes for Guidance"

**Reply by RobDayArup** [reply]
see my reply below

**Comment by hardreddata** [comment]
I am not sure about this.

We recently encountered some mutant _Australian_ AGS data containing LOCA together with GOSA/GORA tables. My learning from this is to hopefully keep the standard simple and clear.

A dedicated flag associated with GEOL_DESC feels like metadata rather than data. Moreover if someone chooses to include comments on soil moisture or consistency in their description per AS1726:2017 examples then this is another case to accommodate (ie: information from outside the GEOL table entering GEOL_DESC). I feel like AGS data producers wont bother to thoughtfully set the flag each time.

Thanks for the presentation last week.

**Reply by RobDayArup** [reply]
Rather than having a boolean switch to say whether the GEOL_DESC was built completely from components the other method I was going to put forward was to have a GEOL_NCDE field to contain a Non-component Description as an extra component field. 

This would make it patently clear to anyone receiving the AGS data that it was generated using the Simplified GEOL and any other component fields are supplemental.  (Note this is how OpenGround software deals with this internally in the AGS4.1AU implementation).  If you consider it as a component field then it also arguably addresses the "components SHALL take precedence over the contents of GEOL_DESC" issue when the Simplified GEOL is used but some additional component fields are populated.  

I initially rejected this version for AGS import/export because it makes the AGS files bigger, since whenever GEOL_NCDE is populated GEOL_DESC would have exactly the same data.  However, this is also essentially true (less formatting) for the Extended GEOL method, so if it makes it clearer for everyone I would be happy to go down this path instead. 

From a contractual/practical perspective I still maintain that GEOL_DESC needs to take precedence over the individual component fields for the reasons discussed in separate response to Phil.

**Comment by RobDayArup** [comment]
Sorry Phil, thanks for pointing this out.  However I still think from both a contractual and practical perspective saying that the individual components SHALL take precedence over the contents of GEOL_DESC is not desirable for the reasons I discussed above.  To restate:
* Creation of the GEOL_DESC requires processing of the component fields and different software will compile this slightly differently
* Many receivers of data may only be able to read the GEOL_DESC field if they are set up to use the "Simplified GEOL"  system and need to be able to rely on this as fact.
* You are effectively saying the GEOL_DESC should not be imported if there are any component fields present, therefore what is the reason for requiring this to be populated?
* It also contradicts the philosophy stated on the Aus Geomechanics web site that  "the individual ‘granular’ headings are optional as part of an extended GEOL group, depending upon the amount of granularity preferred for a particular project."  Since as soon as you enter any of these fields the GEOL_DESC is by definition of the guidance notes to be ignored, but there will not be sufficient information to build the description that is in the AGS file in GEOL_DESC.  Hence it removes the possibility for clients/projects to agree the level of granularity on a case-by-case basis.
